src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -60,6 +70,10 @@ public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidCon
 
                         FluidUtils.setFluidForContainerItem(removedItem, fluidType);
 
+                        // Fill this fluid container up to max capacity.
```

Comment:
Oh,  shoot... I forgot to mention that changing the component without saving it will work in single player,  but will not work in multiplayer scenarios.  Something to look out for.